src/transactions/TransactionTable.js

```diff
@@ -26,30 +27,132 @@ import TransactionForm from './TransactionForm';
-    && (t.date <= filters.endDate
-    || filters.endDate === null
-    || filters.endDate === undefined),
-  ));
+  const [direction, setdirection] = useState(-1);
+  const [sortProperty, setSortProperty] = useState();
+
+  const onButtonClickPayee = () => {
+    setSortProperty('payee');
+    setdirection(-1 * direction);
+  };
+  const onButtonClickDate = () => {
+    setSortProperty('date');
+    setdirection(-1 * direction);
+  };
+  const onButtonClickCategory = () => {
+    setSortProperty('category');
+    setdirection(-1 * direction);
+  };
+  const onButtonClickNotes = () => {
+    setSortProperty('notes');
+    setdirection(-1 * direction);
+  };
+  const onButtonClickAmount = () => {
+    setSortProperty('amount');
+    setdirection(-1 * direction);
+  };
 
+  const transactionList = useSelector(
+    (state) => state.transactions.entities.filter(
+      (t) => (t.categoryId === filters.categoryId
+        || filters.categoryId === null
+        || filters.categoryId === undefined)
+        && (t.payee === filters.payee
+          || filters.payee === null
+          || filters.payee === undefined)
+        && (t.date >= filters.startDate
+          || filters.startDate === null
+          || filters.startDate === undefined)
+        && (t.date <= filters.endDate
+          || filters.endDate === null
+          || filters.endDate === undefined),
+    ).sort(
+      (a, b) => {
+        let x;
+        let y;
+        if (sortProperty === 'payee') {
+          x = a.payee.toLowerCase();
+          y = b.payee.toLowerCase();
+        } else if (sortProperty === 'date') {
+          x = a.date;
+          y = b.date;
+        } else if (sortProperty === 'category') {
+          x = state.categories.entities.find(
+            (category) => category.id === a.categoryId,
+          ).name;
+          // x = state.categories.entities.find((category) => category.id === a.categoryId).name;
+          y = state.categories.entities.find(
+            (category) => category.id === b.categoryId,
+          ).name;
+          // eslint-disable-next-line no-console
+          // console.log((state) => state.categories.entities.map((categoryId) => categoryId.id));
+          // console.log(b.categoryId);
+        } else if (sortProperty === 'notes') {
+          x = a.notes;
+          y = b.notes;
+        } else if (sortProperty === 'amount') {
+          x = a.amount;
+          y = b.amount;
+        }
+        if (x < y) {
+          return -1 * direction;
+        }
+        if (x > y) {
+          return 1 * direction;
+        }
+        return 0;
+      },
+    ),
+  );
+  // eslint-disable-next-line no-console
   return (
     <table className="table table-striped">
       <thead>
         <tr>
-          <th>Date</th>
-          <th>Payee</th>
-          <th>Category</th>
-          <th>Notes</th>
-          <th>Amount</th>
+          <th>
+            <button
+              type="button"
+              className="date"
```

Comment:
We don't want to give this a className of "data". There isn't a CSS class called "data", so it doesn't make sense to add the class here. Same for the other buttons.

Reply:
do I need to just erase that line or do I need to re-name the className to 'sortFilter' or something of that nature?

Reply:
You can just remove that line, or maybe give it the bootstrap classes of "btn btn-link"?